frontend/libs/studio-components/src/components/StudioProfileMenu/StudioProfileMenu.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { type ReactNode, type ReactElement, useState } from 'react';
+import classes from './StudioProfileMenu.module.css';
+import { Divider, DropdownMenu } from '@digdir/designsystemet-react';
+import { StudioPageHeaderButton } from '../StudioPageHeader';
+import { type StudioPageHeaderColor } from '../StudioPageHeader/types/StudioPageHeaderColor';
+import { type StudioPageHeaderVariant } from '../StudioPageHeader/types/StudioPageHeaderVariant';
+
+type StudioProfileMenuItemButton = {
+  type: 'button';
+  onClick: () => void;
+};
+
+type StudioProfileMenuItemLink = {
+  type: 'link';
+  href: string;
+  openInNewTab?: boolean;
+};
+
+export type StudioProfileMenuItem = {
+  action: StudioProfileMenuItemButton | StudioProfileMenuItemLink;
+  itemName: string;
+  isActive?: boolean;
+  hasDivider?: boolean;
+};
+
+export type StudioProfileMenuProps = {
+  triggerButtonText?: string;
+  profileImage: ReactNode;
+  profileMenuItems: StudioProfileMenuItem[];
+  color: StudioPageHeaderColor;
+  variant: StudioPageHeaderVariant;
+  ariaLabelTriggerButton: string;
+  truncateAt?: number;
+};
+
+export const StudioProfileMenu = ({
+  triggerButtonText,
+  profileImage,
+  profileMenuItems,
+  color,
+  variant,
+  ariaLabelTriggerButton,
+  truncateAt,
+}: StudioProfileMenuProps): ReactElement => {
+  const [open, setOpen] = useState(false);
+
+  const truncatedText =
+    triggerButtonText && truncateAt
+      ? triggerButtonText.length < truncateAt
+        ? triggerButtonText
+        : `${triggerButtonText.slice(0, truncateAt)}...`
+      : triggerButtonText;
+
+  const handleToggleMenu = () => {
+    setOpen((isOpen) => !isOpen);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
```

Comment:
I find this comment by Codecov strange as line 181 in the test files covers this 🤔

Reply:
Agreed 😊